crates/wasmtime/src/runtime/vm/sys/windows/mmap.rs

```diff
@@ -10,6 +10,22 @@ use windows_sys::Win32::Foundation::*;
 use windows_sys::Win32::Storage::FileSystem::*;
 use windows_sys::Win32::System::Memory::*;
 
+/// Open a file so that it can be mmap'd for executing.
+///
+/// Open the file with read/execute access and only share for
+/// read. This will enable us to perform the proper mmap below
+/// while also disallowing other processes modifying the file
+/// and having those modifications show up in our address space.
+pub fn open_file_for_mmap(path: &Path) -> Result<File> {
```

Comment:
Creating this function seems a bit messy and I'm wondering if a cross-platform `open` is fine for Windows anyway. We do say that `deserialize_file` is unsafe if the file is modified while in use.

Reply:
Yeah I was suggesting we remove this additional code since we say it's `unsafe` to modify the file. But I'm good leaving it in.

Reply:
IIRC this is required on Windows not necessarily for safety but just to get anything working. I have a vague recollection that "just" doing `File::open` causes some mmap operation down the line to return an error or similar. I could be misremembering though (and CI should turn up any issues if there are some)

Reply:
Gotcha. In general, when we can easily and without imposing overhead protect against bad things happening when an `unsafe` contract is violated, we should do it. So let's leave these bits in, even if it turns out they aren't strictly necessary for correctness.